http-transport-binding.md

```diff
@@ -0,0 +1,338 @@
+Events `contentType` attribute.
+
+#### 3.1.2. Event Data Encoding
+
+The [`data` attribute](#22-data-attribute) byte-sequence is used as the HTTP
+message body.
+
+#### 3.1.3. Metadata Headers
+
+All [Cloud Events][CE] attributes with exception of `contentType` and `data`
+are individually mapped to and from distinct HTTP message headers.
+
+##### 3.1.3.1 HTTP Header Names
+
+The naming convention for the HTTP header mapping of attributes is:
+
+    * Prefixed with "CE-"
+    * Each attribute name's first character is capitalized
+
+Examples:
+
+    * `eventTime` maps to `CE-EventTime`
+    * `eventId` maps to `CE-EventId`
+    * `cloudEventsVersion` maps to `CE-CloudEventsVersion`
+
+##### 3.1.3.2 HTTP Header Values
+
+The value for each HTTP header is constructed from the respective attribute's
+[JSON value][JSON-value] representation, compliant with the [JSON event
+format][JSON-format] specification.
+
+The JSON value MUST NOT contain any unencoded LF (ASCII 0x10) or CR (ASCII
+0x13) characters.
+
+JSON objects and arrays are NOT surrounded with single or double quotes.
+
+#### 3.1.4 Examples
+
+This example shows the *binary* mode mapping of an event with an HTTP POST
+request:
+
+``` text
+POST /someresource HTTP/1.1
+Host: webhook.example.com
+CE-CloudEventsVersion: "0.1"
+CE-EventType: "com.example.someevent"
+CE-EventTime: "2018-04-05T03:56:24Z"
+CE-EventId: "1234-1234-1234"
+CE-Source: "/mycontext/subcontext"
+    .... further attributes ...
+Content-Type: application/json; charset=utf-8
+Content-Length: nnnn
+
+{
+    ... application data ...
+}
+```
+
+This example shows a response containing an event:
+
+``` text
+HTTP/1.1 200 OK
+CE-CloudEventsVersion: "0.1"
+CE-EventType: "com.example.someevent"
+CE-EventTime: "2018-04-05T03:56:24Z"
+CE-EventId: "1234-1234-1234"
+CE-Source: "/mycontext/subcontext"
+    .... further attributes ...
+Content-Type: application/json; charset=utf-8
+Content-Length: nnnn
+
+{
+    ... application data ...
+}
+```
+
+### 3.2. Structured Content Mode
+
+The *structured* content mode keeps event metadata and data together in the
+payload, allowing simple forwarding of the same event across multiple routing
+hops, and across multiple transports.
+
+#### 3.2.1. HTTP Content-Type
+
+The [HTTP `Content-Type`][Content-Type] header is set to the media type of an
+[event format](#14-event-formats).
+
+Example for the [JSON format][JSON-format]:
+
+``` text
+Content-Type: application/cloudevents+json; charset=UTF-8
+```
+
+#### 3.2.2. Event Data Encoding
+
+The chosen [event format](#14-event-formats) defines how all attributes,
+including the `data` attribute, are represented.
+
+The event metadata and data is then rendered in accordance with the event
+format specification and the resulting data becomes the HTTP message body.
+
+#### 3.2.3. Metadata Headers
+
+Implementations MAY include the same HTTP headers as defined for the [binary
```

Comment:
While I think its ok for people to do this, we should make it clear that doing so doesn't remove their requirement to include the same properties in the JSON/body.

Reply:
I'm clarifying that.

Reply:
looks good